### PR TITLE
add online store crud in featurestore workspace

### PR DESCRIPTION
**Pull request by feng-pan8014**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by MayankKumar91 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/feature_store_settings.py [hunk 1]
```diff
@@ -15,22 +15,29 @@
...
     ):
         """
         :keyword compute_runtime:
         :paramtype compute_runtime: ~azure.ai.ml.entities.ComputeRuntime
         :keyword offline_store_connection_name:
         :paramtype offline_store_connection_name: str
+        :keyword online_store_connection_name:
+        :paramtype online_store_connection_name: str
         """
         self.compute_runtime = compute_runtime if compute_runtime else _ComputeRuntime(spark_runtime_version="3.1.0")
```
[the remark this answers is not in the document]
Comment (feng-pan8014) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/feature_store_settings.py]: fixed

**Review by mingweihe — APPROVED** [review]

**Review by seanyao1 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py [hunk 2]
```diff
@@ -231,6 +262,26 @@ def begin_update(
...
+            raise ValidationError("online store type should be redis")
+
+        if online_store and rest_workspace_obj.feature_store_settings.online_store_connection_name:
+            existing_online_Store_connection = self._workspace_connection_operation.get(
+                resource_group,
+                feature_store.name,
+                rest_workspace_obj.feature_store_settings.online_store_connection_name,
+            )
+
+            if existing_online_Store_connection:
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py:275]: Lets make this camel case existing_online_Store_connection --> existing_online_store_connection

Comment (feng-pan8014) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py]: fixed

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py [hunk 3]
```diff
@@ -127,10 +130,31 @@ def get(self, name: str, **kwargs: Dict) -> _FeatureStore:
                     feature_store.offline_store = _MaterializationStore(
                         type=OFFLINE_MATERIALIZATION_STORE_TYPE, target=offline_Store_connection.properties.target
                     )
-                # materialization identity = identity when created through feature store operations
+
+            online_Store_connection = None
```
[the remark this answers is not in the document]
Comment (feng-pan8014) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py]: fixed

**Review by singankit — APPROVED** [review]

**Review by kristapratico — COMMENTED** [review]
Please don't merge this until it is rebased with main

**Comment by feng-pan8014** [comment]
rebase with main on a new PR: https://github.com/Azure/azure-sdk-for-python/pull/29546

**Pull request closed without merging**